### PR TITLE
Add authflow forgot password page

### PR DESCRIPTION
**Pull request by tung2744**
requires https://github.com/authgear/authgear-server/pull/3419
ref #3408 
https://github.com/authgear/authgear-server/assets/34942210/44659982-79de-4cdf-a443-f8dbc311cd87

### CODE REVIEW

File: pkg/auth/handler/webapp/authflow_controller.go [hunk 1]
```diff
@@ -781,6 +780,7 @@ func (c *AuthflowController) checkPath(w http.ResponseWriter, r *http.Request, s
 	if u.Path != r.URL.Path {
 		// We do not know what causes the mismatch.
 		// Maybe x_step was tempered.
+		c.Logger.Error("path mismatch")
```
Comment (louischan-oursky) [inline pkg/auth/handler/webapp/authflow_controller.go:783]: Maybe it should be a warning.

File: pkg/auth/handler/webapp/authflow_forgot_password_success.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	data := make(map[string]interface{})
+
+	baseViewModel := h.BaseViewModel.ViewModel(r, w)
+	viewmodels.Embed(data, baseViewModel)
+
+	return data, nil
+}
+
+func (h *AuthflowForgotPasswordSuccessHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var handlers AuthflowControllerHandlers
```
Comment (louischan-oursky) [inline pkg/auth/handler/webapp/authflow_forgot_password_success.go:48]: We do not actually need data from the screen. Maybe we can skip using AuthflowController here.

**Review by louischan-oursky — COMMENTED** [review]
What does `x_from_step` actually do?

File: pkg/auth/handler/webapp/authflow_forgot_password_success.go (hunk 2 above)
Comment (tung2744) [reply pkg/auth/handler/webapp/authflow_forgot_password_success.go:48]: Using `Controller.HandleStep` here will make sure the user see this page only if his account recovery flow is on the correct step.

**Comment by tung2744** [comment]
> What does x_from_step actually do?

Originally I introduced it for restoring the previous state of the screen after finished the account recovery flow. However I found it is not needed because we always back to the login screen which start a new flow. So let me remove it.

File: pkg/auth/handler/webapp/authflow_enter_password.go [hunk 3]
```diff
@@ -38,6 +38,7 @@ type AuthflowEnterPasswordViewModel struct {
 	PasswordManagerUsername string
 	ForgotPasswordInputType string
 	ForgotPasswordLoginID   string
+	CurrentXStep            string
```
Comment (louischan-oursky) [inline pkg/auth/handler/webapp/authflow_enter_password.go:41]: CurrentXStep is unused. Please remove it.

**Review by louischan-oursky — COMMENTED** [review]
1. The reset password link references `/flows/reset_password` which is interaction-based. I think we need to implement a authflow-based version of it.
2. Have you resolved the issues you mentioned in the last standup meeting?

**Comment by tung2744** [comment]
> The reset password link references /flows/reset_password which is interaction-based. I think we need to implement a authflow-based version of it.

Yes, I will do it in next pr.

> Have you resolved the issues you mentioned in the last standup meeting?

Should be yes.
Websocket problem -> I disable it for all authflow based ui
Redirect URI problem -> Instead of relying on redirect URI, I added a mechanism to navigate to a defined path on flow end.

There will be another pr for reset password flow.

File: pkg/auth/handler/webapp/authflow_enter_password.go (hunk 3 above)
Comment (louischan-oursky) [reply pkg/auth/handler/webapp/authflow_enter_password.go:41]: @tung2744 In case you missed this.

Comment (tung2744) [reply pkg/auth/handler/webapp/authflow_enter_password.go]: @louischan-oursky  Removed, thanks!

**Pull request merged**